src/components/networked-hand-controls.js

```diff
@@ -0,0 +1,356 @@
+/* global THREE, AFRAME, NAF */
+
+/**
+ * Represent any 6dof controller as a hand with live tracked animated gestures and normalized events.
+ *
+ * Updated and modified adaption of https://github.com/aframevr/aframe/blob/master/src/components/hand-controls.js
+ * Designed for use with NAF, but can also be used without NAF.
+ *
+ * Auto-detects appropriate controller.
+ * Handles common events coming from the detected vendor-specific controls.
+ * Loads hand model with gestures that are applied based on the button pressed.
+ * Translate button events to semantic hand-related event names:
+ * (fist, thumbUp, point, hold, open, pistol) + (start, end)
+ * note that the events were incomplete/broken in prior implementation, and therefore a breaking change
+ */
+
+ AFRAME.registerComponent('networked-hand-controls', {
+    schema: {
+      color: { default: 'white', type: 'color' },
+      hand: { type: "string", default: 'left' },
+      handModelStyle: { type: "string", default: 'highPoly', oneOf: ['lowPoly', 'highPoly', 'toon'] },
+      
+      handModelURL: { type: "string", default: '' }, 
+      // ^for specifying a custom model URL; only allowed at init, not via update
+      // (must correspond to existing models and have matching animations to work properly)
+  
+      gesture: { type: "string", default: 'open' },
+      visible: { type: "bool", default: false },
+    },
+    
+    init() {
+      this.Z = Math.random();
+      this.Y[this.Z] = { contact: {}, reverse: false, isContact: false };
+      
+      this.el.object3D.visible = this.data.visible;
+      
+      this.checkLocalHand();
+      if (this.local) {
+        Object.keys(this.buttonEventMap).forEach(evtName => {
+          this.eventFunctionMap[this.data.hand][evtName] = this.handleButton.bind(this, ...this.buttonEventMap[evtName])
+        })      
+        
+        this.el.addEventListener('controllerconnected', () => {
+          this.el.setAttribute('networked-hand-controls', 'visible', true);
+          this.el.object3D.visible = true;
```

Comment:
This line may be unnecessary if your update() method is already handling this.

Reply:
Great point! Original component had this, I later added the update handler for NAF, didn't catch the overlap.